### PR TITLE
Refactor account fetching during sign-up

### PR DESCRIPTION
**Pull request by 0nko**
This PR moves the account-fetching from the sign-up flow to the site picker to avoid users getting stuck if their account can't be fetched after sign up.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/SitePickerViewModel.kt [hunk 1]
```diff
@@ -658,6 +673,7 @@ class SitePickerViewModel @Inject constructor(
         val isProgressDiaLogVisible: Boolean = false,
         val isPrimaryBtnVisible: Boolean = false,
         val isSecondaryBtnVisible: Boolean = false,
+        val isTertiaryBtnEnabled: Boolean = true,
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/SitePickerViewModel.kt]: We are moving the button "create a store" to a bottom sheet, this will need some changes.

Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/SitePickerViewModel.kt]: Oh yeah, I'll get rid of this then. Thanks!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/signup/SignUpRepository.kt [hunk 2]
```diff
@@ -64,17 +63,6 @@ class SignUpRepository @Inject constructor(
...
-        }
-    }
-
-    private suspend fun onTokenUpdatedSuccessfully(): AccountCreationResult {
-        dispatcher.dispatchAndAwait<Void, OnAccountChanged>(
-            AccountActionBuilder.newFetchAccountAction()
-        ).let { accountFetchResult ->
-            return if (accountFetchResult.isError) {
-                WooLog.w(WooLog.T.LOGIN, message = "Error fetching the user: ${accountFetchResult.error.message}")
-                AccountCreationError(UNKNOWN_ERROR)
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/signup/SignUpRepository.kt:77]: I think we should revert this changes after the changes applied [here](https://github.com/woocommerce/woocommerce-android/pull/7705) from Hicham

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/SitePickerViewModel.kt [hunk 3]
```diff
@@ -91,28 +92,37 @@ class SitePickerViewModel @Inject constructor(
...
+                userAvatarUrl = it.avatarUrl
             )
+            sitePickerViewState = sitePickerViewState.copy(userInfo = userInfo)
+
+            if (appPrefsWrapper.getIsNewSignUp()) {
+                triggerEvent(NavigateToStoreCreationEvent)
+            }
         }
+
+        appPrefsWrapper.markAsNewSignUp(false)
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/sitepicker/SitePickerViewModel.kt:113]: Np, I'd add this inside the ` if (appPrefsWrapper.getIsNewSignUp()) {` so we don't have to access disk unnecessarily.

**Review by JorgeMucientes — APPROVED** [review]

**Comment by 0nko** [comment]
Closing this in favor of #7705.
